src/Adapter/Tax/TaxDataProvider.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Tax;
+
+use PrestaShop\PrestaShop\Core\Domain\Tax\DataProvider\TaxChoiceDataProviderInterface;
+use Tax;
+
+/**
+ * Provides legacy data for tax choice provider
+ */
+final class TaxDataProvider implements TaxChoiceDataProviderInterface
+{
+    /**
+     * @var int
+     */
+    private $langId;
+
+    /**
+     * @param int $langId
+     */
+    public function __construct(int $langId)
+    {
+        $this->langId = $langId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTaxes(): array
+    {
+        return Tax::getTaxes($this->langId);
```

Comment:
You're harsh 😛

Reply:
:roll_eyes:

Reply:
As @sarjon mentioned in several comments above, maybe it would be logical to remove TaxDataProvider and its interface and move TaxChoiceProvider to Adapter namespace but it feels inconstant to the way over choice provider are implemented. Wdyt @matks?

Reply:
TaxChoiceProvider must be moved to Adapter. This is a bad strategy but it has been the strategy we followed so far. We might change the strategy at some point, but for now let's keep doing it.

It's easier to navigate a codebase with a few well-known bad design rules rather than a lot of design rules, some good, some bad, which apply to objects which are identical.